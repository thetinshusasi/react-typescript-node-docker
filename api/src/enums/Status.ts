enum Status {
    PENDING = 'PENDING',
    DONE = 'DONE'
}

export default Status;
