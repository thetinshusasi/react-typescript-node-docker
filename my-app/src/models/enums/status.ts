export enum Status {
  PENDING = "PENDING",
  DONE = "DONE",
}
