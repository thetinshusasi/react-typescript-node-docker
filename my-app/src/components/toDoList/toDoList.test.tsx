import React from "react";
import { shallow } from "enzyme";
import ToDoList from "./index";
import { Button, Table } from "react-bootstrap";
import IToDoItem from "../../models/interfaces/todoItem";
import { Status } from "../../models/enums/status";

describe("<ToDoList />", () => {
  it("renders ToDoList", () => {
    const component = shallow(<ToDoList />);
    expect(component).toMatchSnapshot();
    expect(component.find(".message").text()).toEqual(" No items in list");
    expect(component.find(Table).length).toEqual(1);
  });
  it("renders ToDoList with todos", () => {
    const todos: IToDoItem[] = [
      {
        _id: "id",
        title: "title",
        description: "description",
        status: Status.PENDING,
      },
    ];
    const component = shallow(<ToDoList todos={todos} />);
    expect(component.find(Button).length).toEqual(2);
  });
});
