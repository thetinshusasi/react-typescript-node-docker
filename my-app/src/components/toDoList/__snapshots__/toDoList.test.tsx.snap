// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoList /> renders ToDoList 1`] = `
<ForwardRef
  bordered={true}
  hover={true}
  striped={true}
>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        Title
      </th>
      <th>
        Description
      </th>
      <th
        className="text-center"
      >
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        colSpan={4}
      >
        <h3
          className="message"
        >
           
          No items in list
        </h3>
      </td>
    </tr>
  </tbody>
</ForwardRef>
`;

exports[`<ToDoList /> renders ToDoList 2`] = `
<ForwardRef
  bordered={true}
  hover={true}
  striped={true}
>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        Title
      </th>
      <th>
        Description
      </th>
      <th
        className="text-center"
      >
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        title
      </td>
      <td>
        description
      </td>
      <td>
        <div
          className="btnGroup"
        >
          <Button
            active={false}
            className="btn"
            disabled={false}
            onClick={[Function]}
            variant="success"
          >
            Done
          </Button>
          <Button
            active={false}
            className="btn"
            disabled={false}
            onClick={[Function]}
            variant="danger"
          >
            Delete
          </Button>
        </div>
      </td>
    </tr>
  </tbody>
</ForwardRef>
`;
